add granule.py script with granule segmentation functionality (#114)
sunpy/sunkit-image: changelog/114.feature.rst

> Addition of the ``segment`` module and contains functions to segment granulation in images of the solar photosphere.
> The key functionality is contained in the `.segment` function, which takes a Map 
> and a thresholding method then performs the segmentation, with the segmented Map as output.
> Calls to segment look like:
> .. code-block:: python
>     segmented_data = segment(smap, resolution, skimage_method)
> where 
> *``smap`` is the Map containing an optical image of the solar photosphere.
> *``resolution`` is the spatial resolution of the telescope.
> *``skimage_method`` is the base scikit image segmentation technique to use.
> *``mark_dim_centers`` is a boolean flag indicating whether the user would like to mark dim intergranule centers for future exploration.
> We segment into tri-valued images with 0 = intergranule, 1 = granule, 1.5 = brightpoints, and if marking is desired, dim granule middles marked as 0.5.

Reviewer: ```suggestion
Addition of the ``segment`` module and contains functions to segment granulation in images of the solar photosphere.

The key functionality is contained in the `.segment` function, which takes a Map 
and a thresholding method then performs the segmentation, with a segmented Map as the output.

Calls to segment look like:

.. code-block:: python
    segmented_data = segment(smap, resolution, skimage_method)

where 

* ``smap`` is the Map containing an optical image of the solar photosphere.
* ``resolution`` is the spatial resolution of the telescope.
* ``skimage_method`` is the base scikit image segmentation technique to use.
* ``mark_dim_centers`` is a boolean flag indicating whether the user would like to mark dim intergranule centers for future exploration.

We segment into tri-valued images with 0 = intergranule, 1 = granule, 1.5 = bright points, and if marking is desired, dim granule middles marked as 0.5.
```

Author: Sorry! Missed the spaces when I copied in your suggestion.

> Addition of the sunkit_image/segment.py module. This module cntains functions to segment 
> granulation in images of the solar photosphere.
> 
> The key functionality is contained in the `segment` function, which takes in a SunPy map 
> and a thresholding method and performs the segmentation, with the segmented image output 
> as a SunPy map. Calls to segment look like: ::
> 
>     segmented_data = segment(smap, resolution, skimage_method)
> 
> Where 
> 
> *`smap` is a SunPy map containing an optical image of the solar photosphere
> *`resolution` is the spatial resolution of the telescope
> *`skimage_method` is the base sci-kit image (skimage) segmentation technique to use 
> *`mark_dim_centers` is a boleen flag indicating whether the user would like to mark dim 
>   intergranule centers for future exploration.
> 
> We segment into tri-valued images with 0 = intergranule, 1 = granule, 1.5 = faculae, and 
> (if marking is desired) dim granule middles marked as 0.5.

Reviewer: ```suggestion
Addition of the ``segment`` module and contains functions to segment granulation in images of the solar photosphere.

The key functionality is contained in the `.segment` function, which takes a Map 
and a thresholding method then performs the segmentation, with the segmented Map as output.
Calls to segment look like:

.. code-block:: python

    segmented_data = segment(smap, resolution, skimage_method)

where 

*``smap`` is the Map containing an optical image of the solar photosphere.
*``resolution`` is the spatial resolution of the telescope.
*``skimage_method`` is the base scikit image segmentation technique to use.
*``mark_dim_centers`` is a boolean flag indicating whether the user would like to mark dim intergranule centers for future exploration.

We segment into tri-valued images with 0 = intergranule, 1 = granule, 1.5 = faculae, and if marking is desired, dim granule middles marked as 0.5.
```

> Addition of the `sunkit_image.granule` module and contains functions to segment granulation in images of the solar photosphere.

Reviewer 2: ```suggestion
Added the `sunkit_image.granule` which provides functions to segment granulation in images of the solar photosphere.

```

Reviewer 2: I apologize if I am correcting my own corrections here (I know I previously asked for a rewording which may be why some of the language got a bit jumbled).

The key functionality is contained in the `~sunkit_image.granule.segment` function, which 

> segments an image into intergranule, granule,  faculae, and, optionally, dim granule.

Reviewer 2: ```suggestion
segments an image into intergranule, granule, faculae, and, optionally, dim granule.
```
